pkg/api/v1/atlascluster_types.go

```diff
@@ -246,6 +252,43 @@ type ComputeSpec struct {
 	MaxInstanceSize string `json:"maxInstanceSize,omitempty"`
 }
 
+type ProcessArgs mongodbatlas.ProcessArgs
+
+func (specArgs ProcessArgs) IsEqual(newArgs interface{}) bool {
+	specV := reflect.ValueOf(specArgs)
+	newV := reflect.ValueOf(newArgs)
+	typeOfSpec := specV.Type()
+	for i := 0; i < specV.NumField(); i++ {
```

Comment:
do we have nested objects? Is it just top level keys we need to compare?

Reply:
Yeah, thankfully no nested objects in that case, so works as is 😌